awx/ui_next/src/components/Lookup/CredentialsLookup.jsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withI18n } from '@lingui/react';
+import { withRouter } from 'react-router-dom';
+import { t } from '@lingui/macro';
+import { FormGroup, Tooltip } from '@patternfly/react-core';
+import { QuestionCircleIcon as PFQuestionCircleIcon } from '@patternfly/react-icons';
+import styled from 'styled-components';
+
+import { CredentialsAPI, CredentialTypesAPI } from '@api';
+import Lookup from '@components/Lookup';
+
+const QuestionCircleIcon = styled(PFQuestionCircleIcon)`
+  margin-left: 10px;
+`;
+
+class CredentialsLookup extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      credentials: props.credentials,
```

Comment:
`credentials` missing in PropTypes below. Also, consider calling it `initialCredentials` (or `initialValue`?) so it's clear it's not a controlled component (though I would consider making it a controlled component and not setting credentials to state at all)

Reply:
I considered making this a controlled component, but chose not to because I didn't want to add to the state and required functions of `JobTemplateForm`, which is what would control this component.  Additionally, asking `JobTemplateForm` to control this component would also mean passing functions down to its grandchild, which in my opinion, isn't ideal if it can be avoided.